nodewatcher/modules/monitor/sources/snmp/cgm.py

```diff
@@ -0,0 +1,39 @@
+from django import dispatch
+from django.apps import apps
+from django.utils.translation import ugettext as _
+
+from nodewatcher.core.generator.cgm import base as cgm_base
+
+from . import models
+
+
+@cgm_base.register_platform_package('openwrt', 'snmpd', models.SNMPDaemonConfig)
+def snmp_daemon(node, pkgcfg, cfg):
+    """
+    Configures SNMP daemon.
+    """
+
+    try:
+        pkgcfg = pkgcfg.get()
+    except cgm_models.PackageConfig.MultipleObjectsReturned:
+        raise cgm_base.ValidationError(_("Only one SNMP daemon instance can be configured."))
+
+    # Configure SNMP daemon.
+    name = node.config.core.general().get_name()
+
+    system = cfg.snmpd.add('system')
+    system.sysName = name
+    system.sysLocation = pkgcfg.location
+    system.sysContact = pkgcfg.contact
+    system.sysObjectID = node.uuid
+
+    public = cfg.snmpd.add(com2sec='public')
+    public.community = pkgcfg.community
+
+    execute = cfg.snmpd.add(**{'execute': 'exec'})
```

Comment:
I don't think this is correct anymore as you also changed the UCI type. You should leave the dictionary as `**{'exec': 'exec'}`, it is just the variable that has to be `execute` instead of `exec`.